src/QsCompiler/SyntaxProcessor/TypeInference/InferenceContext.fs

```diff
@@ -126,8 +126,10 @@ module private Inference =
     /// True if the characteristics of <paramref name="info1"/> are a subset of the characteristics of
```

Comment:
The function now returns true if either of the arguments are invalid. Could you add that to the description?

Reply:
Though it seems a bit strange for this to return true if either one of these has invalid characteristics. Same for the charsEqual function above.

Reply:
This handling of invalid characteristics is consistent with invalid types, which are treated as a wildcard that match with any other type. That's to avoid too many cascading errors happening when you declare something with an invalid type/characteristics (there should already be an error at the source so we don't need to generate more).

Do you think we should handle or represent this differently?